airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java

```diff
@@ -120,8 +120,8 @@ static boolean isCdc(final JsonNode config) {
   @VisibleForTesting
   static SnapshotIsolation getSnapshotIsolationConfig(final JsonNode config) {
     // new replication method config since version 0.4.0
-    if (config.hasNonNull(REPLICATION_FIELD)) {
-      final JsonNode replicationConfig = config.get(REPLICATION_FIELD);
+    if (config.hasNonNull(LEGACY_REPLICATION_FIELD) && config.get(LEGACY_REPLICATION_FIELD).isObject()) {
```

Comment:
Maybe I should also add `isText` case here too?

Reply:
`.isTextual()` is for when replication_method [was a string](https://github.com/airbytehq/airbyte/commit/90b757b22909a925a04ce172812227ed71f6f27d#diff-09d06a92478799fe02750607b91c60454858f7829e4338e61cc5071fd90d3cf4)
`.isObject()` I think is suitable for a `oneOf` selection field.
Current implementation is with `oneOf` so it sounds like this is correct (?)